firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -84,13 +84,11 @@ private synchronized void executeAllListenerCallbacks(ConfigUpdate configUpdate)
   }
 
   private String parseAndValidateConfigUpdateMessage(String message) {
-    int left = 0;
-    while (left < message.length() && message.charAt(left) != '{') {
-      left++;
-    }
-    int right = message.length() - 1;
-    while (right >= 0 && message.charAt(right) != '}') {
-      right--;
+    int left = message.indexOf('{');
+    int right = message.lastIndexOf('}');
+
+    if (left < 0 || right < 0) {
```

Comment:
unlike the previous test, "}{" is valid. Probably a niche scenario, but worth noting.

Reply:
Yeah I realized that so I still have a check at the bottom of the method in the ternary operator to check if `left >= right` and if its true return nothing so we should be covered for that scenario